include/triton/common/triton_json.h

```diff
@@ -124,9 +124,10 @@ class TritonJson {
 
     // Construct a non-top-level JSON value in a 'document'.
     explicit Value(TritonJson::Value& document, const ValueType type)
```

Comment:
When does this get called? I wonder if adding destructor to the class would solve the issue

Reply:
This constructor is usually used for auto-complete like [this](https://github.com/triton-inference-server/tensorflow_backend/blob/main/src/tensorflow.cc#L1235-L1237).
I tried adding destructor to the class either simply using delete to clean up the Value pointer or using allocator to release Value object. The first case, it will cause double free issue of Value when deleting the DOM object. The second case, the memory leak issue still exists.

Reply:
If the destructor calls `Release()` (which first checks `if value == nullptr` before delete) does this still have a double free issue?

Reply:
No, it won't have a double free issue but it will still have memory leak.